cmd/substreams/init.go

```diff
@@ -172,8 +172,63 @@ func runSubstreamsInitE(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("failed to call discovery endpoint: %w", err)
 		}
 
-		var options []huh.Option[*pbconvo.DiscoveryResponse_Generator]
+		type BlockchainProtocolSelector struct {
+			Id         string
+			Title      string
+			Generators []*pbconvo.DiscoveryResponse_Generator
+		}
+		titleFromGroup := func(group string) string {
```

Comment:
I think this should be extracted to it's own function, to reduce the clutter in the `runSubstreamsInit` a bit. If extracted, I would name it `generatorGroupToProtocolTitle(...)`